src/database/resolvers.js

```diff
@@ -8,22 +8,25 @@ exports.resolver = {
 			args.criteria.lastQueryDate = args.criteria.lastQueryDate || new Date('2001-01-01')
 			args.criteria.lastArticleId = args.criteria.lastArticleId || '000000000000000000000000'
 
-			const articles = await Article.find({
-				link: { $ne: null },
-				modifiedDate: { $gt: new Date(args.criteria.lastQueryDate) },
-				_id: { $gt: args.criteria.lastArticleId },
+			const _categories = Object.values(categories)
+			const promises = _categories.map(async category => {
+				const _articles = await Article.find({
+					category,
+					link: { $ne: null },
+					modifiedDate: { $gt: new Date(args.criteria.lastQueryDate) },
+					_id: { $gt: args.criteria.lastArticleId },
+				})
+					.populate('source')
+					.sort({ _id: -1 })
+					.limit(20)
+				return [..._articles]
 			})
-				.populate('source')
-				.sort({ _id: -1 })
-				.limit(100)
-
-			console.log('Printing articles before sorting', articles.map(a => a.title))
 
-			const sortedArticles = getSortedArticle(articles)
+			const articles = await Promise.all(promises)
 
-			console.log('Printing articles after sorting', sortedArticles.map(a => a.title))
+			const articleFlatterend = _.flatten(articles)
```

Comment:
@shrijan-siris can you write tests around this one?

Reply:
@syuraj i have tried to write the test for resolver , but this is not the required one, i am not getting the articles and just checked whether result is there or not !!

Reply:
@shrijan-siris I think you should be able to require `resolver` and call `getArticles` inside `Query` field for testing. For Unit tests, you can just validate that `Article.find` gets called for each category or not.

Reply:
@shrijan-siris I created a sample unit test here . https://github.com/siristechnology/nepaltoday-api/pull/31/commits/c71d572db95f2d8427b048b8b39b8e9116c0417e

Reply:
<img width="716" alt="Screen Shot 2019-10-22 at 2 59 08 PM" src="https://user-images.githubusercontent.com/54256789/67272788-ec8a9b80-f4dc-11e9-841a-461d93d3d96a.png">
Tried :
- {Article:jest.fn()}
- {Article:jest.fn(mockfile)}

- i think i came the same place where it was started to write the test